### PR TITLE
handle asan check failures

### LINKED ISSUE #343: ASAN log parsing fails to parse "AddressSanitizer CHECK failed" logs
In the linux-radamsa integration test using the ASAN instrumented "trivial-crash" binary, I saw the following log message:

```
unable to parse asan log /tmp/.tmpVZ5iqh/asan-log.13614:
=================================================================
==13614==AddressSanitizer CHECK failed: /build/llvm-toolchain-10-yegZYJ/llvm-toolchain-10-10.0.0/compiler-rt/lib/asan/asan_descriptions.cpp:80 "((0 && "Address is not in memory and not in shadow?")) != (0)" (0x0, 0x0)
    #0 0x49bbee in __asan::AsanCheckFailed(char const*, int, char const*, unsigned long long, unsigned long long) (/onefuzz/blob-containers/oft-setup-3a9daf5d852453d4af5680a55b21d482/fuzz.exe+0x49bbee)
    #1 0x4b010f in __sanitizer::CheckFailed(char const*, int, char const*, unsigned long long, unsigned long long) (/onefuzz/blob-containers/oft-setup-3a9daf5d852453d4af5680a55b21d482/fuzz.exe+0x4b010f)
    #2 0x423c3e in __asan::GetShadowAddressInformation(unsigned long, __asan::ShadowAddressDescription*) (/onefuzz/blob-containers/oft-setup-3a9daf5d852453d4af5680a55b21d482/fuzz.exe+0x423c3e)
    #3 0x424ca6 in __asan::AddressDescription::AddressDescription(unsigned long, unsigned long, bool) (/onefuzz/blob-containers/oft-setup-3a9d...<truncated>
```

There are two issues here.

1. We should differentiate `no error` and `unable to parse error` when handling ASAN log files. 
2. We should fix the support for CHECK failed testing to indicate there is something wrong with the test.

### PR DESCRIPTION
**Pull request by bmc-msft**
Handle parsing "CHECK failed" ASAN logs.  Fixes #343

### CODE REVIEW

File: src/agent/onefuzz/src/asan.rs [hunk 1]
```diff
@@ -20,7 +20,11 @@ pub struct AsanLog {
 
 impl AsanLog {
     pub fn parse(text: String) -> Option<Self> {
-        let (summary, sanitizer, fault_type) = parse_summary(&text)?;
+        let (summary, sanitizer, fault_type) = match parse_summary(&text) {
+            Some(x) => x,
+            None => parse_failure(&text)?,
```
Comment (ranweiler) [inline src/agent/onefuzz/src/asan.rs:25]: In falling back to `parse_failure()`, aren't we throwing away detail?

Should `parse()` really be returning an enum that explicitly encodes the different cases of:
1. We parsed an ASAN log where the _target_ tripped a fault oracle
2. We parsed an ASAN log where there was an internal ASAN error
3. There was no ASAN-specific logging at all

File: src/agent/onefuzz/src/asan.rs [hunk 2]
```diff
@@ -55,6 +59,16 @@ impl AsanLog {
     }
 }
 
+fn parse_failure(text: &str) -> Option<(String, String, String)> {
```
Comment (ranweiler) [inline src/agent/onefuzz/src/asan.rs:62]: Can we rename this to `parse_asan_runtime_error()` or similar, to make the difference between this an "the target code failed" extra obvious?

File: src/agent/onefuzz/src/asan.rs [hunk 3]
```diff
@@ -226,15 +240,28 @@ mod tests {
...
+            (
+                "data/asan-check-failure.txt",
+                "AddressSanitizer",
+                "CHECK failed",
+                12,
+            ),
+            (
+                "data/asan-check-failure-missing-symbolizer.txt",
+                "AddressSanitizer",
+                "CHECK failed",
```
Comment (ranweiler) [inline src/agent/onefuzz/src/asan.rs:252]: Double-checking: this test case just tests that if the symbolizer is missing, our parser doesn't choke on that in the output, and can still extract that the problem was a `CHECK failed`, right?

**Review by ranweiler — APPROVED** [review]

File: src/agent/onefuzz/src/asan.rs (hunk 3 above)
Comment (bmc-msft) [reply src/agent/onefuzz/src/asan.rs:252]: Correct.

File: src/agent/onefuzz/src/asan.rs (hunk 1 above)
Comment (bmc-msft) [reply src/agent/onefuzz/src/asan.rs]: Discussed offline.

**Pull request merged**